lib/reporting/reports/suppliers/helpers/line_items_access_helper.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Reporting
+  module Reports
+    module Suppliers
+      module Helpers
+        module LineItemsAccessHelper
+          def variant(line_items)
+            line_items.first.variant
+          end
+
+          def order(line_items)
+            line_items.first.order
+          end
+
+          def supplier(line_items)
+            variant(line_items).supplier
+          end
+
+          def distributor(line_items)
+            order(line_items).distributor
+          end
+
+          def item_order_cycle(line_items)
+            line_items.first.order_cycle
+          end
+
+          def adjustments_by_type(line_items, type, included: false)
+            total_amount = 0.0
+            adjustment_type = type == :tax ? 'Spree::TaxRate' : 'EnterpriseFee'
+            adjustments = line_items.flat_map(&:adjustments)
+
+            adjustments.each do |adjustment|
+              if adjustment.originator_type == adjustment_type
+                amount = included == adjustment.included ? adjustment.amount : 0.0
+                total_amount += amount
+              end
+            end
```

Comment:
It looks like this code is filtering based on multiple conditions (`originator_type` and `adjustment.included`), using different methods (`if` and a ternary)

Another way would be to use `filter` for both conditions, then `sum`. Hrmm, but that's looping twice 🤔 
I'm probably thinking too much about it.

Reply:
Another thought: the column `total_tax` is calling this twice. 
If `included` could be `nil` and allow both values, then we could do it in one call: `adjustments_by_type(line_items, :tax, included: nil)`

Reply:
Hmm...maybe..😅 I think I'm not getting this right 😅

Reply:
It's fine, this works fine, I just couldn't help by try to optimise it ;)